java-runtime-builder-app/src/main/java/com/google/cloud/runtimes/builder/config/domain/JdkServerLookup.java

```diff
@@ -71,32 +59,28 @@ private void validate() {
    *     valid JDK image is found, an {@link IllegalArgumentException} will be thrown.
    */
   public String lookupJdkImage(String jdk) {
+    Map<String, String> jdkRuntimeMap = getJdkRuntimeMap();
     if (jdk == null) {
       return jdkRuntimeMap.get(KEY_WILDCARD);
     }
 
     if (!jdkRuntimeMap.containsKey(jdk)) {
       throw new IllegalArgumentException(String.format("The provided runtime_config.jdk option '%s'"
-          + " is invalid for JAR deployments. Please use a supported jdk option: %s",
+              + " is invalid for JAR deployments. Please use a supported jdk option: %s",
           Strings.nullToEmpty(jdk), getAvailableJdks()));
     }
     return jdkRuntimeMap.get(jdk);
   }
 
   private List<String> getAvailableJdks() {
-    return jdkRuntimeMap.keySet()
+    return getJdkRuntimeMap().keySet()
         .stream()
         .filter((key) -> !key.contains(KEY_WILDCARD))
         .map(key -> "'" + key + "'")
         .collect(Collectors.toList());
   }
 
-  private String buildServerMapKey(String jdk, String serverType) {
-    String jdkKey = jdk == null ? KEY_WILDCARD : jdk;
-    String serverTypeKey = serverType == null ? KEY_WILDCARD : serverType;
-
-    return jdkKey + KEY_DELIMITER + serverTypeKey;
-  }
+  public abstract Map<String, String> getJdkRuntimeMap();
```

Comment:
This breaks the encapsulation. You shouldn't need the map if you have the lookup method.
Also, you might want to rethink the the use of an abstract class and instead, if needed, introduce an interface.

Reply:
I don't think an interface is necessary, since originally this lookup part was just a single class and it was adequate as that. I did remove these get-map functions in the newest revision.